packages/blocks/src/lib/slam-map-2d/motion-path.svelte

```diff
@@ -15,30 +17,24 @@ import { renderOrder } from './render-order';
 
 extend({ Line2, LineMaterial, LineGeometry });
 
-export let path: string | undefined;
+export let path: Float32Array | undefined;
 
-const updatePath = (pathstr?: string) => {
-  if (pathstr === undefined) {
+const updatePath = (xy?: Float32Array) => {
+  if (xy === undefined) {
     return;
   }
 
   const lineGeometry = new LineGeometry();
-  const points: number[] = [];
-
-  for (const xy of pathstr.split('\n')) {
-    const [xString, yString] = xy.split(',');
-    if (xString !== undefined && yString !== undefined) {
-      const x = Number.parseFloat(xString) / 1000;
-      const y = Number.parseFloat(yString) / 1000;
-      if (Number.isNaN(x) || Number.isNaN(y)) {
-        continue;
-      }
-      points.push(x, y, 0);
+  const xyz: number[] = [];
+  for (let i = 0; i < xy.length - 1; i += 2) {
+    const x = xy[i];
+    const y = xy[i + 1];
+    if (x !== undefined && y !== undefined) {
+      xyz.push(x, y, 0);
     }
   }
 
-  const vertices = new Float32Array(points);
-  lineGeometry.setPositions(vertices);
+  lineGeometry.setPositions(new Float32Array(xyz));
```

Comment:
Ah dang, I completely forgot that `setPositions` requires an `xyz` and not `xy` array.

We can probably find a way to avoid copies altogether in the future but for now we could remove one copy here just by passing a regular JS array:

 ```suggestion
  lineGeometry.setPositions(xyz);
```